Source/WebCore/Modules/highlight/HighlightRegister.cpp

```diff
@@ -40,16 +40,20 @@ void HighlightRegister::initializeMapLike(DOMMapAdapter& map)
 
 void HighlightRegister::setFromMapLike(AtomString&& key, Ref<Highlight>&& value)
 {
+    AtomString k = key;
+    m_orderedHighlightNames.append(WTFMove(k));
```

Comment:
Just use `m_orderedHighlightNames.append(key);` here.

Reply:
Here we are appending, there is no particular sorting / ordering besides the insertion order. I think we can just rename the vector to `m_highlightNames`.

Also, do we need to worry about duplicates here? Should we only append to the vector is the key isn't already in the map?

Reply:
Yes, we do need to worry about duplicates. I forgot to add that check bc it used to be a `ListHashSet` but then I changed it to a vector. Will add that check now!